### PR TITLE
fix(billing): purchase order fix first valid date not selectable

### PR DESCRIPTION
**Pull request by ghyenne**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/MANAGER-6696` 
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | MANAGER-9110 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] ~~Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description

I implemented date management via moment which allows a better management of date calculation and fixed some date selection bug in ui-kit (calendar).

### CODE REVIEW

File: packages/manager/modules/billing/src/orders/purchaseOrders/billing-orders-purchases.service.js [hunk 1]
```diff
@@ -98,7 +98,7 @@ export default class BillingOrdersPurchasesService {
   }
 
   minDateForEndDate() {
-    this.moment = moment();
+    this.moment = moment(null, 'YYYY-MM-DD');
     return this.moment.add(1, 'day').toDate();
```
Comment (mohammed-zahaf) [inline packages/manager/modules/billing/src/orders/purchaseOrders/billing-orders-purchases.service.js:102]: The both lines produce an `Invalid Date`, is't expected behaviour ?

**Review by anooparveti — APPROVED** [review]

File: packages/manager/modules/billing/src/orders/purchaseOrders/billing-orders-purchases.routing.js [hunk 2]
```diff
@@ -61,7 +63,7 @@ export default /* @ngInject */ ($stateProvider) => {
 
       hideBreadcrumb: /* @ngInject */ () => false,
 
-      minDate: /* @ngInject */ () => new Date(),
+      minDate: /* @ngInject */ () => moment().format('YYYY-MM-DD'),
```
Comment (radireddy) [inline packages/manager/modules/billing/src/orders/purchaseOrders/billing-orders-purchases.routing.js:66]: 'YYYY-MM-DD' is used in multiple places. Can you extract it as a constant.

File: packages/manager/modules/billing/src/orders/purchaseOrders/billing-orders-purchases.service.js (hunk 1 above)
Comment (ghyenne) [reply packages/manager/modules/billing/src/orders/purchaseOrders/billing-orders-purchases.service.js]: Indeed, in my tests it came up false. I was misled by the result. I made some changes.

File: packages/manager/modules/billing/src/orders/purchaseOrders/billing-orders-purchases.routing.js (hunk 2 above)
Comment (ghyenne) [reply packages/manager/modules/billing/src/orders/purchaseOrders/billing-orders-purchases.routing.js]: It's applied

**Review by vks-ovh — APPROVED** [review]

**Pull request merged**